### ISSUE TITLE
Command crashes Editor on macOS release

### ISSUE THREAD

**Issue by CrossR** [onivim/oni2#594]
Using `:split` / `:vsplit` crashes the editor on macOS, as well as some other commands such as `:terminal` and assumedly a bunch more.

What is odd is that these crashes are very reproducible for me in the Release when running from Spotlight, but not at all when running directly. That is opening Oni2 via Finder/Dock/Spotlight causes a crash, but launching it directly via `/Applications/Onivim2.App/Contents/MacOS/Oni2` I do not get a crash. To my knowledge, these are both the exact same binaries, so its not even an issue with Release vs Source.

I am running a few day old build, but this has been confirmed on discord too.

**Comment by CrossR** [comment]
(Out of interest I gave the Ripgrep issue a quick try too and that is reproducible in both direct and spotlight launches, so I think this is a separate bug to that.)

**Comment by bryphe** [comment]
Thanks @CrossR for the details! Looking into this now.

Given the way it manifests (opening via Finder/Dock/Spotlight, but not w/ the executable directly) - I suspect that it might be related to the 'hardened runtime' that we enable for our signed builds.

**Comment by bryphe** [comment]
Further investigation - it's not related to the codesigning or hardened runtime at all. I was able to reproduce on a package on `master` with `esy create-release`.

It seems that it _only_ reproduces with `:vsp` or `:sp` against the untitled buffer. If I run those commands with a filename passed, or `:vsp` with a buffer that isn't the untitled buffer - it works OK.

**Comment by bryphe** [comment]
Seems like what's happening is we're not handling the 'path' correctly for the untitled buffer - I turned on file logging + debug logging and see this:
```
[ERROR] Exception Invalid_argument("\"\": invalid path"):
Raised at file "stdlib.ml", line 34, characters 20-45
Called from file "src/Path.re", line 16, characters 2-12
Called from file "src/editor/Model/LanguageInfo.re", line 32, characters 2-18
Called from file "src/editor/UI/Tab.re", line 96, characters 6-75
```
```

**Comment by CrossR** [comment]
Ah so this issue is the same as #579? There is also a PR open for that (#580) though I think that is to avoid the issue rather than fixing why we are unable to open that.

**Comment by bryphe** [comment]
It's related (it seems like there are a class of issues where we are crashing trying to read `"\"` as a path), but a bit different.

The flow for this is pretty weird! On Windows, for example, I noticed if I do `:vsp` on the untitled buffer, i get:
![vsp-bug](https://user-images.githubusercontent.com/13532591/62496266-422c9300-b78d-11e9-9164-fdfff81f0360.gif)

I should have two `untitled` buffers, but I ended up with an `untitled` and an `oni2`. It turns out, when we call `Vim.Buffer.openFile` with an empty file... it defaults to the current directory. For my scenario above, I ran oni2 from `D:/oni2`, so it was given that name (and shows `oni2` in the UI).

However, in the open-from-finder scenarion on Mac, the default working directory is `/` - so it uses that as the name for the buffer.

A simple fix for this is to give an explicit name to the buffer, so when we split, we don't end up in that default-use-directory-case.

An alternative, and probably better fix, would be, in the case where no file is specified for vsp/sp, we create a split based on the buffer _number_ (we always use the path right now). I went with the simpler fix for #596 

And it turns out this issue reproduces by running Onivim 2 from the development repo on Mac, and then running `:cd /`, and running `:vsp` on the first buffer.
